packages/sanity/src/core/form/studio/assetSourceMediaLibrary/hooks/useProvision.ts

```diff
@@ -0,0 +1,202 @@
+  | 'ERROR_NO_MEDIA_LIBRARIES_FOUND'
+  | 'ERROR_NO_ORGANIZATION_FOUND'
+  | 'ERROR_FAILED_TO_CREATE_MEDIA_LIBRARY'
+
+export class ProvisionError extends Error {
+  message: string
+  error: string
+  code: ProvisionErrorCode
+  // eslint-disable-next-line unicorn/custom-error-definition
+  constructor(message: string, error: string, code: ProvisionErrorCode) {
+    // eslint-disable-next-line unicorn/custom-error-definition
+    super(message)
+    this.message = message
+    this.error = error
+    this.code = code
+  }
+}
+
+type ProvisionResponse = {
+  id?: string
+  organizationId?: string
+  status: 'provisioning' | 'active' | 'loading' | 'error'
+  error?: ProvisionError
+}
+
+function getMediaLibrariesForOrganization(
+  client: ObservableSanityClient,
+  organizationId: string,
+): Observable<MediaLibrary[]> {
+  return client
+    .request({
+      uri: `/media-libraries?organizationId=${organizationId}`,
+      method: 'GET',
+    })
+    .pipe(
+      concatMap((data) => {
+        if (data && Array.isArray(data.data)) {
+          return of(data.data)
+        }
+        throw new ProvisionError(
+          'No media libraries found',
+          'No media libraries found',
+          'ERROR_NO_MEDIA_LIBRARIES_FOUND',
+        )
+      }),
+    )
+}
+
+function getOrganizationIdFromProjectId(
+  client: ObservableSanityClient,
+  projectId: string,
+): Observable<string> {
+  return client
+    .request({
+      uri: `/projects/${projectId}`,
+      method: 'GET',
+    })
+    .pipe(
+      concatMap(async (data) => {
+        if (data.organizationId) {
+          return data.organizationId
+        }
+        throw new ProvisionError(
+          'Organization ID not found',
+          'Organization ID not found',
+          'ERROR_NO_ORGANIZATION_FOUND',
+        )
+      }),
+    )
+}
+
+function requestMediaLibraryStatus(
+  client: ObservableSanityClient,
+  libraryId: string,
+): Observable<ProvisionResponse> {
+  return client
+    .request({
+      uri: `/media-libraries/${libraryId}`,
+      method: 'GET',
+    })
+    .pipe(
+      concatMap((data) => {
+        if (data && data.id) {
+          return of({
+            id: data.id,
+            organizationId: data.organizationId,
+            status: data.status,
+          } satisfies ProvisionResponse)
+        }
+        throw new ProvisionError(
+          'Media library not found',
+          'Media library not found',
+          'ERROR_NO_MEDIA_LIBRARIES_FOUND',
+        )
+      }),
+    )
+}
+
+function createMediaLibrary(
+  client: ObservableSanityClient,
+  organizationId: string,
+): Observable<ProvisionResponse> {
+  return client.request({
+    uri: `/media-libraries`,
+    method: 'POST',
+    body: {organizationId},
+  })
+}
+
+export function useProvision(projectId: string): ProvisionResponse {
+  if (!projectId) {
+    throw new Error('projectId is required to fetch organizationId')
+  }
+  const client = useClient({apiVersion: 'vX'}).observable
```

Comment:
These APIs should be available on `2025-02-19`
```suggestion
  const client = useClient({apiVersion: 'v2025-02-19'}).observable
```